From faaac88f0686066e0b930952f12010f6d93fd6cf Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Mon, 4 Oct 2021 19:41:34 +0200
Subject: [PATCH] mount: make mount units start jobs not runnable if
 /p/s/mountinfo ratelimit is in effect

(cherry picked from commit a7c93dfe91e88a5a561341c523a45c7f8d71a588)

Related: #2037395
---
 src/core/mount.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/core/mount.c b/src/core/mount.c
index 72341fe685..dbac1b7cb1 100644
--- a/src/core/mount.c
+++ b/src/core/mount.c
@@ -1953,6 +1953,9 @@ static int mount_can_start(Unit *u) {
 
         assert(m);
 
+        if (sd_event_source_is_ratelimited(u->manager->mount_event_source))
+                return -EAGAIN;
+
         r = unit_test_start_limit(u);
         if (r < 0) {
                 mount_enter_dead(m, MOUNT_FAILURE_START_LIMIT_HIT);
